app/models/host/managed.rb

```diff
@@ -184,7 +184,8 @@ class Jail < ::Safemode::Jail
     if fromtime.nil? || totime.nil?
       raise ::Foreman.Exception.new(N_("invalid time range"))
     else
-      joins("INNER JOIN reports ON reports.host_id = hosts.id").where("reports.reported_at BETWEEN ? AND ?", fromtime, totime)
+      joins("INNER JOIN reports as run_distribution_reports ON run_distribution_reports.host_id = hosts.id")
```

Comment:
hmm, while this makes sense, why not just use `.joins(:all_reports)` and let rails take care of the naming?

Reply:
just for information we have encounter this bug in production with the 18.3 version and  this fix have worked for us.